kafka/consumer.go

```diff
@@ -329,15 +329,20 @@ func (c *Consumer) rebalance(ev Event) bool {
 // consumerReader reads messages and events from the librdkafka consumer queue
 // and posts them on the consumer channel.
 // Runs until termChan closes
-func consumerReader(c *Consumer, termChan chan bool) {
+func consumerReader(c *Consumer, termChan chan bool, eventsChanSize int) {
 
 out:
 	for true {
 		select {
 		case _ = <-termChan:
 			break out
 		default:
-			_, term := c.handle.eventPoll(c.events, 100, 1000, termChan)
+			_, term := c.handle.eventPoll(
+				c.events,
+				100,
+				eventsChanSize - len(c.events), // max events depends on available buffer
```

Comment:
Which would be the preferred way to handle this? sleep for 100ms, default to `eventsChanSize`, or poll for a single event?